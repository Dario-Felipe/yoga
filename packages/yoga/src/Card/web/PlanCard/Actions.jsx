import styled from 'styled-components';

import Actions from '../Card/Actions';
import theme from '../../../Theme/helpers/themeReader';

const PlanActions = styled(Actions)`
  margin-top: ${theme.components.card.plan.actions.margin.top}px;
`;

PlanActions.displayName = 'PlanCard.Actions';

export default PlanActions;
