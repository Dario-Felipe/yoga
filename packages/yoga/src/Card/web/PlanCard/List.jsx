import React from 'react';
import styled, { css } from 'styled-components';
import { string, node, shape, oneOfType, func } from 'prop-types';

import theme from '../../../Theme/helpers/themeReader';

const truncateStyle = css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const List = styled.ul`
  margin: ${theme.components.card.plan.list.margin.top}px 0 0;
  padding: 0;

  list-style: none;
`;

const Item = styled.li`
  ${truncateStyle}

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      margin-bottom: ${plan.list.item.margin.bottom}px;

      font-size: ${plan.list.item.font.size}px;
      color: ${plan.list.item.font.color};

      svg {
        margin-right: ${plan.list.item.icon.margin.right}px;
      }
    `;
  }}
`;

const Button = styled.button`
  display: block;
  padding: 0;

  letter-spacing: normal;

  background-color: transparent;
  border: none;

  cursor: pointer;
  text-decoration: none;

  ${truncateStyle}

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      margin-top: ${plan.list.button.margin.top}px;

      font-family: ${theme.baseFont.family};
      font-size: ${plan.list.button.font.size}px;
      font-weight: ${plan.list.button.font.weight};
      color: ${plan.list.button.font.color};
    `;
  }}
`;

const ListItem = ({ text, icon: Icon, buttonProps }) => (
  <Item>
    {Icon && <Icon width={16} height={16} />}
    {text}
    {Object.keys(buttonProps).length && <Button {...buttonProps} />}
  </Item>
);

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: string.isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
};

ListItem.defaultProps = {
  icon: undefined,
  buttonProps: {},
};

export { List, ListItem };
