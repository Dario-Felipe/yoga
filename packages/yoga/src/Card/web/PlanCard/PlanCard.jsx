import React from 'react';
import styled, { css } from 'styled-components';
import { node, oneOf, string } from 'prop-types';
import Box from '../../../Box';
import Text from '../../../Text';

export const PLAN_LINE_HEIGHT = 4;

const Plan = styled.article`
  width: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
  max-width: 312px;

  overflow: hidden;

  ${props => {
    const theme = props.theme.yoga;
    const { plan } = theme.components.card;

    return css`
      padding: ${plan.padding.top}px ${plan.padding.right}px
        ${plan.padding.bottom}px ${plan.padding.left}px;
      border: ${theme.borders.small}px solid ${theme.colors.light};
      border-radius: ${plan.radius}px;

      background-color: ${theme.colors.white};
    `;
  }}
`;

const Border = styled.span`
  position: absolute;
  top: 0;
  left: 0;

  display: inline-block;
  width: 100%;

  height: ${PLAN_LINE_HEIGHT}px;
  background-color: ${({
    variant,
    theme: {
      yoga: {
        colors: { deepPurple, [variant]: color = deepPurple },
      },
    },
  }) => color};
`;

const DiscountWrapper = styled(Box).attrs({
  ph: 'xsmall',
  pv: 'xxxsmall',
})`
  ${({ bg, theme }) => {
    const MIN_WRAPPER_HEIGHT = 28;
    const baseCurveStyle = css`
      content: '';
      position: absolute;
      left: -${MIN_WRAPPER_HEIGHT}px;
      height: ${MIN_WRAPPER_HEIGHT}px;
      width: ${MIN_WRAPPER_HEIGHT}px;
    `;

    return css`
      position: absolute;
      top: 0;
      right: 0;
      display: flex;
      align-items: center;
      min-height: ${MIN_WRAPPER_HEIGHT}px;
      border-bottom-left-radius: 16px;

      ::before {
        ${baseCurveStyle}
        top: 0;
        background: ${theme.yoga.colors[bg]};
      }

      ::after {
        ${baseCurveStyle}
        top: ${PLAN_LINE_HEIGHT}px;
        background: ${theme.yoga.colors.white};
        border-top-right-radius: ${MIN_WRAPPER_HEIGHT / 2 - PLAN_LINE_HEIGHT}px;
      }
    `;
  }}
`;

function PlanCard({ children, discount, variant, ...rest }) {
  return (
    <Plan {...rest}>
      {discount && (
        <DiscountWrapper bg={variant}>
          <Text.Tiny color="white" fontWeight="medium">
            {discount}
          </Text.Tiny>
        </DiscountWrapper>
      )}
      <Border variant={variant} />
      {children}
    </Plan>
  );
}

PlanCard.propTypes = {
  children: node,
  discount: string,
  /** style the card border top color following the theme (primary, secondary,
   * vibin, hope, energy, relax, peace, verve, uplift, deepPurple, deep,
   * stamina, dark, medium, light, clear, white) */
  variant: oneOf([
    'primary',
    'secondary',
    'vibin',
    'hope',
    'energy',
    'relax',
    'peace',
    'verve',
    'uplift',
    'deepPurple',
    'stamina',
    'dark',
    'medium',
    'deep',
    'light',
    'clear',
    'white',
  ]),
};

PlanCard.defaultProps = {
  children: undefined,
  discount: undefined,
  variant: 'deepPurple',
};

PlanCard.displayName = 'PlanCard';

export default PlanCard;
