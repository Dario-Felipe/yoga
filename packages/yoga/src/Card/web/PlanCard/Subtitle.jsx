import styled from 'styled-components';

import theme from '../../../Theme/helpers/themeReader';
import Text from '../../../Text';

const Subtitle = styled(Text.Tiny)`
  margin-bottom: 8px;

  color: ${theme.components.card.plan.subtitle.color};
  font-weight: 500;

  text-transform: uppercase;
`;

Subtitle.displayName = 'PlanCard.Subtitle';

export default Subtitle;
