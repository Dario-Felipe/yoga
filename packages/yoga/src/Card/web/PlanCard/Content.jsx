import React from 'react';
import styled, { css } from 'styled-components';
import { string, node } from 'prop-types';

import theme from '../../../Theme/helpers/themeReader';
import Text from '../../../Text';
import Content from '../Card/Content';
import Subtitle from './Subtitle';
import Divider from '../../../Divider';

const Wrapper = styled(Content)`
  flex: 1;
`;

const Title = styled(Text.Medium)`
  ${props => {
    const {
      components: { card, cardweb },
    } = theme(props);

    return css`
      color: ${card.plan.title.color};
      margin-bottom: ${cardweb.plan.title.margin.bottom}px;
      line-height: ${cardweb.plan.title.lineHeight}px;
    `;
  }}
`;

const Description = styled(Text.Small)`
  height: 40px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;

  color: ${theme.components.card.plan.description.color};

  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
`;

const Price = styled.div`
  display: flex;

  ${props => {
    const {
      components: { cardweb },
    } = theme(props);

    return css`
      margin-bottom: ${cardweb.plan.price.margin.bottom}px;
    `;
  }}
`;

const EnhancePrice = styled(Text.Medium)`
  align-self: center;
`;

const PlanCardContent = ({
  title,
  subtitle,
  description,
  currency,
  suffix,
  price,
  period,
  children,
  ...rest
}) => (
  <Wrapper {...rest}>
    {subtitle && <Subtitle>{subtitle}</Subtitle>}
    {title && <Title>{title}</Title>}
    {description && <Description>{description}</Description>}
    <Price>
      {currency && <Text.H4 as="sup">{currency}</Text.H4>}
      {price && <Text.H4 as="strong">{price}</Text.H4>}
      {suffix && <Text.H4 as="sup">{suffix}</Text.H4>}
      {period && <EnhancePrice as="sub">{period}</EnhancePrice>}
    </Price>
    <Divider />
    {children}
  </Wrapper>
);

PlanCardContent.propTypes = {
  /** plan name */
  title: string.isRequired,
  /** currency of the current country */
  currency: string,
  /** suffix currency of the current country */
  suffix: string,
  price: string.isRequired,
  /** period that this price will be charged  */
  period: string.isRequired,
  description: string,
  subtitle: string,
  children: node,
};

PlanCardContent.defaultProps = {
  children: null,
  description: null,
  subtitle: null,
  currency: null,
  suffix: null,
};

PlanCardContent.displayName = 'PlanCard.Content';

export default PlanCardContent;
