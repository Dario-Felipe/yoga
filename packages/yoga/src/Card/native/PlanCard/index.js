import PlanCard from './PlanCard';
import Content from './Content';
import Actions from './Actions';
import Subtitle from './Subtitle';
import { List, ListItem } from './List';

PlanCard.Content = Content;
PlanCard.Actions = Actions;
PlanCard.Subtitle = Subtitle;
PlanCard.List = List;
PlanCard.ListItem = ListItem;

export default PlanCard;
