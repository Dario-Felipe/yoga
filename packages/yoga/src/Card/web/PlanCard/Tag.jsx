import styled, { css } from 'styled-components';

import { PLAN_LINE_HEIGHT } from './PlanCard';
import Tag from '../../../Tag';

const StyledTag = styled(Tag.Informative).attrs(() => ({ small: true }))`
  position: absolute;
  display: inline-flex;
  align-items: center;
  justify-content: center;

  ${props => {
    const { plan } = props.theme.yoga.components.card;

    return css`
      top: ${plan.tag.position.top + PLAN_LINE_HEIGHT}px;
      left: ${plan.tag.position.left}px;
    `;
  }}
`;

StyledTag.displayName = 'PlanCard.Tag';

export default StyledTag;
