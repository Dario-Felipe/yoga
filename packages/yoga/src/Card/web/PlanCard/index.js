import PlanCard from './PlanCard';
import Content from './Content';
import Actions from './Actions';
import Subtitle from './Subtitle';
import Tag from './Tag';
import { List, ListItem } from './List';

PlanCard.Content = Content;
PlanCard.Actions = Actions;
PlanCard.Subtitle = Subtitle;
PlanCard.Tag = Tag;
PlanCard.List = List;
PlanCard.ListItem = ListItem;

export default PlanCard;
