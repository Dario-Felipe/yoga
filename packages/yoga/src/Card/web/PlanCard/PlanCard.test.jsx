import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Star, MapPin } from '@gympass/yoga-icons';
import { ThemeProvider, Button, Icon } from '../../..';
import PlanCard from '.';

describe('<PlanCard />', () => {
  const buttonOnClickMock = jest.fn();

  const renderPlan = ({ clickableItems } = {}) =>
    render(
      <ThemeProvider>
        <PlanCard>
          <PlanCard.Tag variant="informative">Recommended Plan</PlanCard.Tag>
          <PlanCard.Content
            subtitle="plan"
            title="Basic"
            currency="$"
            price="99.90"
            period="/month"
          >
            <PlanCard.Subtitle>Get access to</PlanCard.Subtitle>
            <PlanCard.List>
              <PlanCard.ListItem
                icon={
                  <Icon
                    as={MapPin}
                    height="small"
                    width="small"
                    stroke="medium"
                  />
                }
                text="gyms and studios"
                onClick={clickableItems && jest.fn()}
              />
              <PlanCard.ListItem
                icon={Star}
                text="list item"
                buttonProps={{
                  children: 'button',
                  as: 'a',
                  onClick: buttonOnClickMock,
                }}
                onClick={clickableItems && jest.fn()}
              />
            </PlanCard.List>
          </PlanCard.Content>
          <PlanCard.Actions>
            <Button full>Select this plan</Button>
          </PlanCard.Actions>
        </PlanCard>
      </ThemeProvider>,
    );

  describe('Events', () => {
    it('should call onClick when ListItem has a button', () => {
      const { getByText } = renderPlan();

      fireEvent.click(getByText('button'));

      expect(buttonOnClickMock).toHaveBeenCalled();
    });

    it('should make list item clickable', () => {
      const itemClickMock = jest.fn();

      const { getByRole } = render(
        <ThemeProvider>
          <PlanCard variant="energy">
            <PlanCard.Content title="Basic">
              <PlanCard.List>
                <PlanCard.ListItem
                  icon={
                    <Icon
                      as={MapPin}
                      height="small"
                      width="small"
                      stroke="medium"
                    />
                  }
                  text="gyms and studios"
                  onClick={itemClickMock}
                />
              </PlanCard.List>
            </PlanCard.Content>
          </PlanCard>
        </ThemeProvider>,
      );

      fireEvent.click(getByRole('button', { label: 'gyms and studios' }));

      expect(itemClickMock).toHaveBeenCalled();
    });
  });

  describe('Snapshots', () => {
    it('should match snapshot with default PlanCard', () => {
      const { container: planCard } = renderPlan();

      expect(planCard).toMatchSnapshot();
    });

    it('should match snapshot with variant', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard variant="hope">
            <PlanCard.Content
              subtitle="plan"
              title="Basic"
              currency="$"
              price="99.90"
              period="/month"
            />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with suffix', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard variant="hope">
            <PlanCard.Content
              subtitle="plan"
              title="Basic"
              currency="$"
              suffix="MXN"
              price="99.90"
              period="/month"
            />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should render the extra content', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard>
            <PlanCard.Content title="Basic" extra={<p>Hello</p>} />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should render the title without a colored badge', () => {
      const { getByText } = render(
        <ThemeProvider>
          <PlanCard>
            <PlanCard.Content title="Basic" />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(getByText('Basic')).toMatchSnapshot();
    });

    it('should render the title with a colored badge', () => {
      const { getByText } = render(
        <ThemeProvider>
          <PlanCard>
            <PlanCard.Content title="Basic" badgeColor="energy" />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(getByText('Basic')).toMatchSnapshot();
    });

    it('should render the discount', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard discount="SAVE 40%" variant="energy">
            <PlanCard.Content title="Basic" />
          </PlanCard>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should render list item content as button', () => {
      const { container } = renderPlan({ clickableItems: true });

      expect(container).toMatchSnapshot();
    });
  });
});
