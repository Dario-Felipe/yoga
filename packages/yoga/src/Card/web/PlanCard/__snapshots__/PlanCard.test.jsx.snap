// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanCard /> Snapshots should match snapshot with default PlanCard 1`] = `
.c15 {
  box-sizing: border-box;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  width: 100%;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #F46152;
  border: 2px solid #F46152;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 700;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15:not([disabled]):hover,
.c15:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(244,97,82,0.45);
}

.c15:active {
  background-color: #F78D82;
  border-color: #F78D82;
  color: #FFFFFF;
}

.c15:disabled {
  background-color: #DBDBE8;
  border-color: #DBDBE8;
  color: #9898A6;
  cursor: not-allowed;
}

.c8 {
  margin: 0;
  padding: 0;
  font-size: 48px;
  line-height: 56px;
  font-weight: 500;
  font-family: Rubik;
  color: #41414A;
}

.c0 {
  width: 100%;
  display: inline-block;
  position: relative;
  max-width: 288px;
  overflow: hidden;
  padding: 64px 20px 32px 20px;
  border-radius: 16px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  display: inline-block;
  width: 100%;
  height: 8px;
  background-color: #710252;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  text-transform: uppercase;
  margin-bottom: 8px;
  color: #6B6B78;
  font-weight: 500;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 28px;
  font-weight: 500;
  font-family: Rubik;
  color: #41414A;
  color: #001027;
  margin-bottom: 4px;
}

.c5 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  height: 40px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  color: #6B6B78;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 32px;
  margin-bottom: 40px;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c7 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 4px;
}

.c14 {
  margin-top: 24px;
}

.c2 {
  position: absolute;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  top: 28px;
  left: 20px;
  padding-left: 8px;
  padding-right: 8px;
  border: 1px solid #9898A6;
  border-radius: 4px;
  font-size: 12px;
  line-height: px;
  font-weight: 500;
}

.c10 {
  margin: 16px 0 0;
  padding: 0;
  list-style: none;
}

.c11 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 16px;
}

.c11 svg {
  vertical-align: middle;
  margin-right: 12px;
}

.c12 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  display: inline-block;
  height: 16px;
  vertical-align: middle;
  color: #001027;
}

.c13 {
  display: block;
  padding: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: transparent;
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 12px;
  font-family: Rubik;
  font-size: 12px;
  font-weight: 500;
  color: #652CDB;
}

<div>
  <article
    class="c0"
  >
    <span
      class="c1"
    />
    <span
      class="c2"
    >
      Recommended Plan
    </span>
    <div
      class=""
    >
      <p
        class="c3"
      >
        plan
      </p>
      <h5
        class="c4"
      >
        Basic
      </h5>
      <p
        class="c5"
      />
      <div
        class="c6"
      >
        <sup
          class="c7"
        >
          $
        </sup>
        <strong
          class="c8"
        >
          99.90
        </strong>
        <sub
          class="c9"
        >
          /month
        </sub>
      </div>
      <p
        class="c3"
      >
        Get access to
      </p>
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          <svg
            fill="#9898A6"
            height="16"
            width="16"
          />
          <span
            class="c12"
          >
            list item
          </span>
          <a
            class="c13"
          >
            button
          </a>
        </li>
      </ul>
    </div>
    <footer
      class="c14"
    >
      <button
        class="c15"
      >
        Select this plan
      </button>
    </footer>
  </article>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot with variant 1`] = `
.c7 {
  margin: 0;
  padding: 0;
  font-size: 48px;
  line-height: 56px;
  font-weight: 500;
  font-family: Rubik;
  color: #41414A;
}

.c0 {
  width: 100%;
  display: inline-block;
  position: relative;
  max-width: 288px;
  overflow: hidden;
  padding: 64px 20px 32px 20px;
  border-radius: 16px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  display: inline-block;
  width: 100%;
  height: 8px;
  background-color: #1D856C;
}

.c2 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  text-transform: uppercase;
  margin-bottom: 8px;
  color: #6B6B78;
  font-weight: 500;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 28px;
  font-weight: 500;
  font-family: Rubik;
  color: #41414A;
  color: #001027;
  margin-bottom: 4px;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  height: 40px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  color: #6B6B78;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 32px;
  margin-bottom: 40px;
}

.c8 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #41414A;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 4px;
}

<div>
  <article
    class="c0"
  >
    <span
      class="c1"
    />
    <div
      class=""
    >
      <p
        class="c2"
      >
        plan
      </p>
      <h5
        class="c3"
      >
        Basic
      </h5>
      <p
        class="c4"
      />
      <div
        class="c5"
      >
        <sup
          class="c6"
        >
          $
        </sup>
        <strong
          class="c7"
        >
          99.90
        </strong>
        <sub
          class="c8"
        >
          /month
        </sub>
      </div>
    </div>
  </article>
</div>
`;
